### DISCUSSION TITLE
Unexpected server routing instead of client while using next/link

### DISCUSSION THREAD

**Discussion by march08** [vercel/next.js#52797] [Help]
### Summary

Hi, we are encountering some unexpected server routing (probably?) when `_document` render function is triggered and with that whole _app client is rerendered. This however still maintains the state of e.g. `react-query` 

This is happening only on this route (file) defined in `/pages` (from another to this route or within this route specification)
```
/w/:workspaceId/events/:eventId/[[...subpage]].tsx
```

### Additional information

* Using latest next version 13.4.10
* We are leveraging some `/app` features (transitioning) however the error while navigating is from /pages routes to the one described above
* If added `shallow` property to the links/push, there is a correct client side behaviour. 
* routeChangeStart is not triggered 

When this happens? (Using either `next/link` or `useRouter().push()`)

* `/w/dx-heroes/events` -> `/w/dx-heroes/events/830f9c7e-1624-4e54-9469-b31094d21e99/registrations`
* `/w/dx-heroes/events/830f9c7e-1624-4e54-9469-b31094d21e99/registrations`-> `/w/dx-heroes/events/830f9c7e-1624-4e54-9469-b31094d21e99/resources`
* `any page` -> `/w/:workspaceId/events/:eventId/[[...subpage]]`

There might be some high level issue or missuse from our end, but I am not even sure what to look for, any help or hints, sharing experience is appreciated!

**Comment by icyJoseph** [comment]
Could you clarify some things:

- You are using both app and pages routers?
- the issue happens when a link goes from app router to pages router?

**Reply by horaklukas** [reply]
I can clarify as I'm co-developer 🙂 

* yes, we started process of incremental migration and are currently with some routes in `pages` and some in `app`
* this is the weirdest part, all the listed routes ☝️  are inside the `pages` and doesn't cooperate with any route from the `app` dir

**Reply by icyJoseph** [reply]
Aha, but it is not like it's navigating from app dir to pages dir, because that I'd expect to cause a page reload.

Could you show us a simplified tree of the app/pages folder?

**Reply by horaklukas** [reply]
No it's not, the `app` router routes are quite isolated now

```
app/
   BaseLayout.tsx
   (legal-docs)
        layout.tsx
        terms-and-conditions/
            page.tsx
    (embed)
        layout.tsx
        not-found.tsx
        w-p/
           [workspaceId]/
               page.tsx
pages/
    w/
      [workspaceId]/
         events/
            index.tsx
            [eventId]/
                 [[...subpage]].tsx
```

it's something like that ☝️ I included only relevant routes and few routes from app dir

**Reply by horaklukas** [reply]
Not sure If you're still on it, I'm sharing some updates about it. We found out that the hard reload is forced also for another page in `pages` dir when navigating between pages, eg. from `events` to `resources`

```
pages/
   company/
      [slug]/
         [slug].tsx
         events.tsx
         resources.tsx
         join.tsx
```

and I had a bit of time so I tried to migrate this page and all its subpages into the `app` dir and suddenly the hard reload disappear and.
I also first I tried to migrate from `next/router` to `next/navigation` while still keeping the page in the `pages` dir, but it didn't help with the issue.

So to me it looks like there is an issue in Next.js routing when trying to incrementally adopt new `app` architecture with all the new hooks and routing utils... 😬
